src/core/components/text-area/styles.ts

```diff
@@ -8,6 +8,7 @@ import { resets } from '@guardian/src-foundations/utils';
 export const errorInput = css`
 	border: 4px solid ${border.error};
 	color: ${text.error};
+	margin-top: 0;
```

Comment:
I'm assuming this is necessary because of the order in which [styles are applied](https://github.com/guardian/source/blob/64bfef4f3b41c4e2177179a6d50373f5ee598058/src/core/components/text-area/TextArea.tsx#L109-L110) to the `textarea` element. If we move the `errorInput` style above the `supportingTextMargin` / `labelMargin` style, does this negate the need for the `margin-top` here?

Reply:
Well, it's actually because there's already a lot of whitespace in the `errorInput` and `successInput`, so we discussed that for now a margin should not be applied if the error or success is visible as it already looks like there is a margin there.

![chrome_4pcYtnaWTG](https://user-images.githubusercontent.com/13315440/135609019-e7bfc190-56ff-49fd-98e7-b77e8daaf45f.png)

Which means that we have to explicitly negate the margins that we applied using the `supportingTextMargin` / `labelMargin`.

Reply:
I see! You could say it's a CSS override to avoid a nested ternary in the JSX... caught between a rock and a hard place 😅  I think your way is best 👍